pandas/core/internals/concat.py

```diff
@@ -335,9 +335,13 @@ def _concatenate_join_units(join_units, concat_axis, copy):
         # the non-EA values are 2D arrays with shape (1, n)
         to_concat = [t if isinstance(t, ExtensionArray) else t[0, :] for t in to_concat]
         concat_values = concat_compat(to_concat, axis=0)
-        if not isinstance(concat_values, ExtensionArray):
+        if not isinstance(
+            concat_values, ExtensionArray
+        ) or concat_values.dtype == np.dtype("M8[ns]"):
```

Comment:
@jbrockmendel what's the best way to check for naive DatetimeArray? (so only numpy datetime64[ns]) Eg `is_datetime64_ns_dtype` still gives True for datetime64[ns, tz]

Reply:
`isinstance(arr, DatetimeArray) and arr.tz is None`

Reply:
Hmm, is that necessarily better than what I did above?

Reply:
fine by me

Reply:
don't really have a preference, mostly want to be consistent with other code, so if the snippet you showed is used elsewhere, will change it